MSAL/src/MSALPublicClientApplication.m

```diff
@@ -244,25 +244,28 @@ - (nullable instancetype)initWithConfiguration:(nonnull MSALPublicClientApplicat
     
 #if TARGET_OS_IPHONE
     // Optional Paramater
-    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:config.cacheConfig.keychainSharingGroup];
+    MSIDKeychainTokenCache *dataSource = [[MSIDKeychainTokenCache alloc] initWithGroup:config.cacheConfig.keychainSharingGroup error:&msidError];
+    
+    if (!dataSource)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to initialize iOS keychain cache.");
```

Comment:
nit: would it be useful to log the error here as well? Or we already log it in keychain layer? Would like to make sure we have enough diagnostics based on logs.

Reply:
We are currently logging both places which should be fine.